pandas/core/internals/blocks.py

```diff
@@ -643,8 +652,8 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
                 )
         return newb
 
-    def _astype(self, dtype: DtypeObj, copy: bool) -> ArrayLike:
-        values = self.values
+    @staticmethod
```

Comment:
can you move to module level instead (or maybe to pandas.core.dtypes.cast); -1 on using staticmethod

Reply:
sure.  and as is traditional ill point out: this is yet another kludge that wont be necessary with 2D EAs